### PR TITLE
Add upkeepID logging to each step in job pipeline

### PR DESCRIPTION
**Pull request by begmaroman**
- Added UpkeepID to the peform upkeep tx metadata.
- Updated logging logic in the tx manager. 
- Upgraded integrations framework.

### CODE REVIEW

File: core/chains/evm/txmgr/eth_broadcaster.go [hunk 1]
```diff
@@ -365,6 +365,27 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
...
+		"checker", etx.TransmitChecker,
+		"gasLimit", etx.GasLimit,
+		"gasPrice", attempt.GasPrice,
+		"gasTipCap", attempt.GasTipCap,
+		"gasFeeCap", attempt.GasFeeCap,
+	)
+	if meta, err := etx.GetMeta(); err == nil && meta != nil {
+		lgr = lgr.With(
+			"jobID", meta.JobID,
+			"upkeepID", meta.UpkeepID,
```
Comment (infiloop2) [inline core/chains/evm/txmgr/eth_broadcaster.go:379]: would this default to 0 for non keeper jobs? Is there a way we can avoid adding this for non keeper jobs

**Review by infiloop2 — COMMENTED** [review]
lgtm overall. One question to check if we can avoid polluting non keeper logs

**Comment by infiloop2** [comment]
Also just a general comment, would be good to know how you tested this change. (Added tests / Relied on existing tests / Locally tested)

**Comment by begmaroman** [comment]
@infiloop2 I tested it by running keepers locally to make sure everything works as expected. There were no business logic changes, only logging logic updates.

File: core/chains/evm/txmgr/eth_broadcaster.go (hunk 1 above)
Comment (begmaroman) [reply core/chains/evm/txmgr/eth_broadcaster.go:379]: Good question. Upkeep ID starts from 0 so I'm not sure. Let me think about it.

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably eb23b7b81b02428a5e47e9c0c5591aeb1b568474) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/chains/evm/txmgr/eth_broadcaster.go (hunk 1 above)
Comment (begmaroman) [reply core/chains/evm/txmgr/eth_broadcaster.go:379]: Fixed

File: core/chains/evm/txmgr/models.go [hunk 2]
```diff
@@ -31,6 +33,8 @@ type EthTxMeta struct {
 	// Used for the VRFv2 - the subscription ID of the
 	// requester of the VRF.
 	SubID uint64 `json:"SubId"`
```
Comment (infiloop2) [inline core/chains/evm/txmgr/models.go:35]: Not related to this change, just curious, this field seems specific to vrf, is it included in all logs with some default value?

**Review by infiloop2 — APPROVED** [review]

File: core/chains/evm/txmgr/models.go (hunk 2 above)
Comment (begmaroman) [reply core/chains/evm/txmgr/models.go]: Was thinking about that as well. Updated that logic.

**Review by infiloop2 — DISMISSED** [review]
If not already done, could you test locally with a vrf job to ensure logs go smoothly and there are no nil panics

File: core/chains/evm/txmgr/eth_broadcaster.go [hunk 3]
```diff
@@ -375,47 +375,46 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
...
 	checkCtx, cancel := context.WithTimeout(ctx, TransmitCheckTimeout)
 	defer cancel()
-	err = checker.Check(checkCtx, eb.logger, etx, attempt)
+	err = checker.Check(checkCtx, lgr, etx, attempt)
 	if errors.Is(err, context.Canceled) {
-		eb.logger.Errorw("Transmission checker timed out, sending anyway",
-			"ethTxId", etx.ID,
-			"meta", etx.Meta,
-			"checker", etx.TransmitChecker)
+		lgr.Error("Transmission checker timed out, sending anyway")
```
Comment (essamhassan) [inline core/chains/evm/txmgr/eth_broadcaster.go:393]: According to the new [logging guidelines](https://github.com/smartcontractkit/chainlink/blob/develop/core/logger/logger.go), this seems to be matching the description for warning especially that we handle it gracefully and send anyway

File: core/chains/evm/txmgr/eth_broadcaster.go [hunk 4]
```diff
@@ -375,47 +375,46 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
 		return errors.Wrap(err, "building transmit checker")
 	}
 
+	lgr, err := etx.GetLogger(eb.logger.With(
+		"gasPrice", attempt.GasPrice,
+		"gasTipCap", attempt.GasTipCap,
+		"gasFeeCap", attempt.GasFeeCap,
+	))
+	if err != nil {
+		eb.logger.Errorw("failed to get tx logger meta of the transaction", "err", err)
```
Comment (essamhassan) [inline core/chains/evm/txmgr/eth_broadcaster.go:384]: qq: what happens here for the rest of the func call if logger errors and lggr is nil? not sure but probably need to fallback if eth meta is missing

File: core/chains/evm/txmgr/eth_broadcaster.go [hunk 5]
```diff
@@ -375,47 +375,46 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
...
+	}
```
Comment (makramkd) [inline core/chains/evm/txmgr/eth_broadcaster.go:385]: In the case where `err != nil`, would `lgr` be nil? Should you do `lgr = eb.logger.With(...)` in this case as a fall back?

File: core/chains/evm/txmgr/eth_broadcaster.go [hunk 6]
```diff
@@ -375,47 +375,46 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
...
+		lgr.Error("Transmission checker timed out, sending anyway")
 	} else if err != nil {
 		etx.Error = null.StringFrom(err.Error())
-		eb.logger.Infow("Transmission checker failed, fatally erroring transaction.",
-			"ethTxId", etx.ID,
-			"meta", etx.Meta,
-			"checker", etx.TransmitChecker,
-			"err", err)
-		return eb.saveFatallyErroredTransaction(&etx)
+		lgr.Infow("Transmission checker failed, fatally erroring transaction.", "err", err)
```
Comment (essamhassan) [inline core/chains/evm/txmgr/eth_broadcaster.go:396]: This seems to need to be bumped to warn imo based on [logging guidelines](https://github.com/smartcontractkit/chainlink/blob/develop/core/logger/logger.go), I think I missed this one in my audit

File: core/chains/evm/txmgr/models.go [hunk 7]
```diff
@@ -16,21 +16,25 @@ import (
...
-	JobID         int32
-	RequestID     common.Hash
-	RequestTxHash common.Hash
+	JobID         int32       `json:"JobID"`
+	RequestID     common.Hash `json:"RequestID"`
+	RequestTxHash common.Hash `json:"RequestTxHash"`
 	// Used for the VRFv2 - max link this tx will bill
 	// should it get bumped
-	MaxLink string
+	MaxLink *string `json:"MaxLink,omitempty"`
```
Comment (makramkd) [inline core/chains/evm/txmgr/models.go:32]: Why change this type to be a pointer to string?

Comment (begmaroman) [reply core/chains/evm/txmgr/models.go:32]: This is the VRF specific field. Going to be added to the logger only if a TX is related to VRF.

File: core/chains/evm/txmgr/models.go [hunk 8]
```diff
@@ -16,21 +16,25 @@ import (
...
 	// Used for the VRFv2 - the subscription ID of the
 	// requester of the VRF.
-	SubID uint64 `json:"SubId"`
+	SubID *uint64 `json:"SubId,omitempty"`
```
[the remark this answers is not in the document]
Comment (begmaroman) [reply core/chains/evm/txmgr/models.go:35]: Same answer

File: core/chains/evm/txmgr/eth_broadcaster.go (hunk 4 above)
Comment (begmaroman) [reply core/chains/evm/txmgr/eth_broadcaster.go]: Fixed

File: core/chains/evm/txmgr/eth_broadcaster.go (hunk 3 above)
Comment (begmaroman) [reply core/chains/evm/txmgr/eth_broadcaster.go]: Fixed

**Review by essamhassan — APPROVED** [review]

**Pull request merged**